### ISSUE TITLE
[Accessibility][ConfigPanelThresholdsColorPicker] The modal dialog does not follow the established design pattern 

### ISSUE THREAD

**Issue by mnischay** [awslabs/iot-app-kit#2365]
#### Notes and Advice 
The "Color Picker" dialog component has the following issues:
 
The screen reader does not announce the name and role information for the dialog (fails [SC 4.1.2 Name, Role, Value](https://www.w3.org/TR/WCAG/#name-role-value)).

####  Recommendation 
Ensure that screen readers should announce the name and role information for the dialog (fails [SC 4.1.2 Name, Role, Value](https://www.w3.org/TR/WCAG/#name-role-value)).
 
For dialogs without a visible title, use aria-label to provide an accessible name.
Add the role dialog OR alertdialog to the containing element.
 
For Example: Conditionally providing polyfill for inert attribute
 
 
if (typeof(document.body.querySelector('*').inert) === 'undefined') {
    // need to polyfill inert for firefox desktop/mobile, samsung internet and some older browser versions
    // see https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/inert#browser_compatibility
    console.debug("Support for inert attribute not detected, adding polyfill");
    // If using package manager:
    // await import("wicg-inert")
    // or 
    // Vanilla js method
    // Replace source and integrity attributes as needed
    let script = document.createElement("script");
    script.setAttribute("src", "https://unpkg.com/wicg-inert");
    script.setAttribute("type", "module")
    // integrity of wicg-inert@3.1.2
    script.setAttribute("integrity", "sha256-YUdsOmqbJZlyMtzVHbNCUm9dEeRRbTJ5SxaZlnhdgOE=")
    script.setAttribute("async", "true");
    script.setAttribute("crossorigin", "anonymous")
    document.body.appendChild(script);
}
 #### Resources 
[ARIA Authoring Practices Modal Dialog Example](https://w3c.github.io/aria-practices/examples/dialog-modal/dialog.html)
[Accessible Modal Dialog](https://github.com/scottaohara/accessible_modal_window)
[a11y-dialog widget](https://github.com/KittyGiraudel/a11y-dialog)
[The current state of modal dialog accessibility](https://www.tpgi.com/the-current-state-of-modal-dialog-accessibility/)
[WICG inert polyfill](https://github.com/WICG/inert)
[MDN for inert attribute](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/inert)
[HTML standard inert attribute](https://html.spec.whatwg.org/multipage/interaction.html#dom-inert)
[Steve Faulkner, aria-hasPopUp less is more](https://html5accessibility.com/stuff/2023/06/20/aria-haspopup-less-is-more/)

**Comment by Kausar-HM** [comment]
Suggestion is - screen readers should announce the name and role information for the dialog.
But aria-label is already present, screen reader can read it.
If we add role as 'dialog' OR 'alertdialog', we get an error 'Interactive elements should not be assigned non-interactive roles'.

Existing input element:
```
<input
  aria-label='color picker'
  type='color'
  value={color}
  onChange={(e) => {
   updateColor(e.target.value);
  }}
  className='color-picker'
  style={{ backgroundColor: color }}
  {...other}
/> 
```

**Comment by jmbuss** [comment]
Spoke with someone from the accessibility team. The problem is that the dialog which launches on click does not announce its role / name upon opening. This is a problem with how the color input is handled natively by the browser. I have proposed a new solution to UX which we are still working through together.

**Comment by jmbuss** [comment]
Confirmed with accessibility team that we no longer need to do this. Closing the ticket.
